homeassistant/components/media_player/denon.py

```diff
@@ -153,6 +161,8 @@ def update(self):
         else:
             self._mediainfo = self.source
 
+        self._sound_mode = self.telnet_request(telnet, 'PSDYNVOL ?')
```

Comment:
All these telnet commands should be extracted into an external lib and hosted on PyPI. We cannot take any PRs for Denon that introduce new Telnet commands.

Reply:
In that case I believe it will be better for me to work on expanding scarface's denonavr http platform as it already exists since the telnet stuff is a bit hacky.  I will start work on an av_receiver component that handles an A/V Receiver rather than handling media playback.  If you agree then we can close this pull request.  Thanks!